bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/AddonSuggestionFinderService.java

```diff
@@ -0,0 +1,153 @@
+    private final LocaleProvider localeProvider;
+
+    @Activate
+    public AddonSuggestionFinderService(@Reference LocaleProvider localeProvider) {
+        this.localeProvider = localeProvider;
+        loadSupportingAddon();
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void addAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
+        if (!addonInfoProviders.contains(addonInfoProvider)) {
+            addonInfoProviders.add(addonInfoProvider);
+            scanStart();
+        }
+    }
+
+    public void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
+        if (addonInfoProviders.contains(addonInfoProvider)) {
+            addonInfoProviders.remove(addonInfoProvider);
+            scanStart();
+        }
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void addAddonSuggestionFinder(AddonSuggestionFinder addonSuggestionFinder) {
+        if (!addonSuggestionFinders.contains(addonSuggestionFinder)) {
+            addonSuggestionFinders.add(addonSuggestionFinder);
+            scanStart();
+        }
+    }
+
+    public void removeAddonSuggestionFinder(AddonSuggestionFinder addonSuggestionFinder) {
+        if (addonSuggestionFinders.contains(addonSuggestionFinder)) {
+            addonSuggestionFinders.remove(addonSuggestionFinder);
+            scanStart();
+        }
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void addAddonService(AddonService addonService) {
+        addonServices.add(addonService);
+    }
+
+    public void removeAddonService(AddonService addonService) {
+        addonServices.remove(addonService);
+    }
+
+    @Deactivate
+    @Override
+    public void close() throws Exception {
+        addonSuggestionFinderTasks.forEach(t -> t.cancel(true));
+        addonSuggestionFinderTasks.clear();
+        addonSuggestionFinders.forEach(f -> f.reset());
+        addonSuggestionFinders.clear();
+        addonInfoProviders.clear();
+        addonServices.clear();
+    }
+
+    public List<Addon> getSuggestedAddons(@Nullable Locale locale) {
+        Set<String> uids = addonSuggestionFinders.stream().map(f -> f.getAddonSuggestionUIDs())
+                .flatMap(Collection::stream).collect(Collectors.toSet());
+
+        return addonServices.stream().map(s -> s.getAddons(locale)).flatMap(Collection::stream)
+                .filter(a -> uids.contains(a.getUid())).toList();
+    }
+
+    /**
+     * Loads our special supporting addon 'org.openhab.misc.addonsuggestionfinder' that implements an AddonInfoProvider
+     * service that provides information about potential available addons that could be suggested to the user to be
+     * installed during setup.
+     */
+    private void loadSupportingAddon() {
```

Comment:
Do we need this? I don't think this should depend on this specific one being installed. Any discovery information from any addonInfoProvider should be used.
The special addon should get installed with the distribution or from the UI.

Reply:
Indeed. I will wait for the responses to the post above.

Reply:
Removed